pkg/metrics/sampler/matcher.go

```diff
@@ -277,70 +262,76 @@ func (ne constantMatcher) String() string {
 	return fmt.Sprint(ne.value)
 }
 
-// NewSampleMatchFn creates new includeSampleMatchFn func, enableProcessMetrics might be nil when
-// value was not set.
-func NewSampleMatchFn(enableProcessMetrics *bool, metricsMatchers config.MetricsMap, ffRetriever feature_flags.Retriever) MatcherFn {
-	// configuration option always takes precedence over FF and matchers configuration
-	if enableProcessMetrics == nil {
-		matcher := matcherFromMetricsMatchers(metricsMatchers)
-		if matcher != nil {
-			return matcher
+// NewIncludeSampleMatchFn returns a function `func(sample) bool` that determinesif the sample
+// should be included (true) as an event or not (false). Note that this is NOT the negation
+// of `NewExcludeSampleMatchFn`.
```

Comment:
Maybe we can also add a comment that it only works for ProcessSample.

Reply:
Done in 1a9f8d8501ca2233fd670f80e9ac238a102cc271!